### PR TITLE
Unquote keys in import attributes

### LINKED ISSUE #15563: Remove quotes in import attribute
**Prettier 3.0.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBLAtgBwgJxgAmAF98AzHCdfAcmvwHdUYALQ-AHRBgE9M5OknAFYBnaJ3xEQAGhARMMVNBHJQAQxwV6ABQ0IVKNQBt6a7itkAjHGrABrODADKa9HAAyqKHGSljIuCsbe0cnTFsvAHNkGBwAV0CQAPRUGPjEkSijOABFOIh4X39E0QAPJyzc-MKkPyMA2QBHarhtCkwDEDURAFpvOAATQZkuG1QjKIBhSnQ1ZC6jIxHMqEjsgEEYWNRLOPhtOBxPbyL6xOYYdCMAdWYmOBFwsDgnfSZUADcmbnmwEQsQB8EgBJKBDWBOMA4VAKdZgpw8bJpBKyTAUALXGyYeZoh6HD4+WReAJ4NpqSKzU4NEDhHAk+aWNSWOBLVHQ2DXVADFjIAAcAAZZDg4M1UMKyRS5rVirIYEzOdzmMgAEyyOIBAAqTIMdWpcHQzIGQwG7jUqzi5LgADFcLMtlF5mo9hAQEQiEA)

```sh
--parser babel
```

**Input:**

```jsx
import {} from '' with { "type":"json" }
```

**Output:**

```jsx
import {} from "" with { "type": "json" };

```

**Expected behavior:**

```jsx
import {} from "" with { type: "json" };
```

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Since resolving conflicts is annoying, I created a new PR.

Closes #15564
Fixes #15563

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Review by fisker — CHANGES_REQUESTED** [review]
Need update login in https://github.com/prettier/prettier/blob/5c6450db9cd456e7442c5c578a18c064c47ce189/src/language-js/print/property.js#L15,  number/bigint can't be unquoted.  https://github.com/prettier/prettier/issues/15563#issuecomment-1784203602

**Review by fisker — APPROVED** [review]
Bug with ~`--prose-wrap always`~ `--quote-props consistent`

**Prettier pr-15888**
[Playground link](https://deploy-preview-15888--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBLAtgBwgJxgAgCsBnafAMxwnXwB0QA6AenQgBMGTp78B3VGAAt8wOiBgBPTHHpIxXKDwC+AbhAAaEBEwxU0YslABDHFV4AFEwgMojAG15GJBzQCMcRsAGs4MAMpG6HAAMqhQcMjk9sRwbh7evn6YnmEA5sgwOACusSAx6KgZ2blwAB7SOBgIMPYAKnA4UCaocDZRdjGaxGl2cACKWRDwkdG5JKV+Pf2Dw0jtnSAAjjNw5lSYNuD6qMTwiJqZRqh2aQDC1OhGyCD2dhp5UwCCMJmorlnw5g2h4SMduYIYOg7AB1QQCVrJMBwPzWASoABuAgk1zAxBcIAROQAklA2NU-GBKjpHni-JJen8FpgqDEQR5MNd7I5nPcwjE8GsjKlLlTcskcBzrpJpMQiagdPcaWEYCDUGwhMgABwABk0ODgy1QGq5PKuc1GByMrjlCsEyAATJosjFasa2oaQHB0K44Gx8WxgkYoKksty4AAxXCXF5pJkfCAgJRKIA)

```sh
--parser typescript
--quote-props consistent
--prose-wrap always
```

**Input:**

```tsx
import json from "./mod.json" with { "type": "json" };
```

**Output:**

```tsx
TypeError: Cannot read properties of undefined (reading 'some')
    at ft (https://deploy-preview-15888--prettier.netlify.app/lib/plugins/estree.js:36:29755)
    at hs (https://deploy-preview-15888--prettier.netlify.app/lib/plugins/estree.js:36:30369)
    at Ta (https://deploy-preview-15888--prettier.netlify.app/lib/plugins/estree.js:36:58094)
    at Vl (https://deploy-preview-15888--prettier.netlify.app/lib/plugins/estree.js:36:72252)
    at Object.Hl (https://deploy-preview-15888--prettier.netlify.app/lib/plugins/estree.js:36:72660)
    at Yn (https://deploy-preview-15888--prettier.netlify.app/lib/standalone.js:27:1809)
    at D (https://deploy-preview-15888--prettier.netlify.app/lib/standalone.js:27:1629)
    at s (https://deploy-preview-15888--prettier.netlify.app/lib/standalone.js:27:1437)
    at https://deploy-preview-15888--prettier.netlify.app/lib/standalone.js:15:4072
    at bt.each (https://deploy-preview-15888--prettier.netlify.app/lib/standalone.js:15:3975)
```

**Review by fisker — CHANGES_REQUESTED** [review]
Approved by mistake

File: tests/format/js/import-attributes/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -265,6 +505,69 @@ exports[`quoted-keys.js [meriyah] format 1`] = `
...
+import json from "./mod.js" with { type: "json" };
+import json from "./mod.js" with { foobar: "json" };
+import json from "./mod.js" with { "type-foo": "json" };
+import json from "./mod.js" with { "333foo": "json" };
+import json from "./mod.js" with { "0": "json" };
+import json from "./mod.js" with { "0n": "json" };
+import json from "./mod.js" with { "33333": "json" };
+import json from "./mod.js" with { "33333n": "json" };
+import json from "./mod.js" with { "33333b": "json" };
+import json from "./mod.js" with { "03333": "json" };
```
Comment (fisker) [inline tests/format/js/import-attributes/__snapshots__/jsfmt.spec.js.snap:566]: I think we need more tests for different `--quote-props` options.

Comment (sosukesuzuki) [reply tests/format/js/import-attributes/__snapshots__/jsfmt.spec.js.snap:806]: [b78e90f](https://github.com/prettier/prettier/pull/15888/commits/b78e90f25226056de6c869411e7aabaac402f940)

**Review by fisker — APPROVED** [review]

**Comment by fisker** [comment]
Something is not right... I'll fix.

**Comment by kachkaev** [comment]
Related issue: https://github.com/prettier/prettier/issues/16340

**Pull request merged**